compiler/damlc/daml-stdlib-src/DA/Text.daml

```diff
@@ -205,14 +205,21 @@ dropWhile p = implode . P.dropWhile p . explode
 dropWhileEnd : (Text -> Bool) -> Text -> Text
 dropWhileEnd p = implode . L.dropWhileEnd p . explode
 
+-- Separate function because recursion is only allowed at the top-level and we
+-- don't want to pay for many implode/explode operations.
+-- This shoud not be exported.
+splitOnHelper : [Text] -> [Text] -> [Text]
+splitOnHelper s [] = [""]
```

Comment:
Apologies for my slow thinking. Seeing the new version in front of me made me realize that we don't need this line either. The line below has the same outcome for `t = ""`.
```suggestion
```

Reply:
What about performance? Is that something I should worry about at all here? This case does less work but I suppose it adds a check to each recursive call, so my guess would be removing this case is probably a win overall.

Reply:
The only call which we slow down as the final call where all the calls in the "hot loop" will have one less pattern matching to perform. Thus, performance should not be impacted negatively.